presto-main/src/main/java/io/prestosql/server/security/oauth2/OAuth2Config.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.security.oauth2;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.validation.FileExists;
+import io.airlift.units.Duration;
+import io.airlift.units.MinDuration;
+
+import javax.validation.constraints.NotNull;
+
+import java.io.File;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class OAuth2Config
+{
+    private Optional<String> stateKey = Optional.empty();
+    private String serverUrl;
+    private String authUrl;
+    private String tokenUrl;
+    private String jwksUrl;
+    private String clientId;
+    private String clientSecret;
+    private Duration challengeTimeout = new Duration(15, TimeUnit.MINUTES);
+    private Optional<String> userMappingPattern = Optional.empty();
+    private Optional<File> userMappingFile = Optional.empty();
+
+    public Optional<String> getStateKey()
+    {
+        return stateKey;
+    }
+
+    @Config("http-server.authentication.oauth2.state-key")
+    @ConfigDescription("A secret key used by HMAC algorithm to sign the state parameter")
+    public OAuth2Config setStateKey(String stateKey)
+    {
+        this.stateKey = Optional.ofNullable(stateKey);
+        return this;
+    }
+
+    @NotNull
+    public String getServerUrl()
+    {
+        return serverUrl;
+    }
+
+    @Config("http-server.authentication.oauth2.server-url")
+    @ConfigDescription("URL of the authorization server")
```

Comment:
How is this different than the other URLs? Can you clarify this with something like the below (replacing it with the proper description)

> URL of the authorization server's main UI page

Reply:
It wasn't being used since we've added jwks url, removed.